app/controllers/volunteers_controller.rb

```diff
@@ -85,7 +85,7 @@ def add_repairs
   def add_repair_to_volunteer
     # @repair = Repair.find(params[:id])
     volunteer_repair = VolunteerRepair.new
-    if current_user.nil?
+    if current_user.onwer.nil?
```

Comment:
This is still a bug.  It should say, if current_user.volunteer.nil?  because you are using the current_user.volunteer.id below.  The latter line would throw an error if volunteer is nil.  Sorry to get hung up on a small point, but it is important in making the application robust.  This is just a one line change.